apps/website/.vuepress/theme/global-components/DocIconDetail.vue

```diff
@@ -116,50 +116,26 @@ export default {
     },
     activateVariant: function (variant) {
       this.activeVariant = variant;
-      this.fetchIconUrl();
-    },
-    hasSolid: function (variants) {
-      const solidTest = variants.filter(v => v.includes('solid'));
-      return solidTest.length > 0 ? true : false;
-    },
-    createSolidDownload: function (variants) {
-      const hasBadge = variants.filter(v => v.includes('has-badge'));
-      const hasAlert = variants.filter(v => v.includes('has-alert'));
-      if (variants.length === 1) {
-        // is solid
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-solid`;
-      } else if (hasBadge.length === 1) {
-        // is solid + badged
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-solid-badged`;
-      } else if (hasAlert.length === 1) {
-        // is solid + alerted
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-solid-alerted`;
-      }
-    },
-    createOutlineDownload: function (variants) {
-      const hasBadge = variants.filter(v => v.includes('has-badge'));
-      const hasAlert = variants.filter(v => v.includes('has-alert'));
-
-      if (hasBadge.length === 1) {
-        // is oulined + badged
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-outline-badged`;
-      } else if (hasAlert.length === 1) {
-        // is outlined + alerted
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-outline-alerted`;
-      }
     },
     fetchIconUrl: function (activeVariant) {
-      const variants = this.activeVariant.classes ? this.activeVariant.classes.filter(variant => variant) : [];
-      if (variants.length === 0) {
-        // not solid, no badge, no alert
-        return `${ICON_DOWNLOAD_URL}set=${this.iconSetName}&shape=${this.iconName}-line`;
-      } else if (this.hasSolid(variants)) {
-        // is a solid icon download, may have badge or alert
-        return this.createSolidDownload(variants);
+      let shape = `&shape=${this.iconName}`;
+
+      // If there any keys in the activeVariant that means is solid or have badge so `line` will not work
+      if (Object.keys(activeVariant).length) {
+        shape += tactiveVariant.solid ? `-solid` : `-outline`;
       } else {
-        // is an outline, has either badge or alert
-        return this.createOutlineDownload(variants);
+        shape += '-line';
```

Comment:
Are there any icons with `-line` suffixes? IIRC, default line icons have `-outline` suffix, right?

Reply:
It's seems not 

```bash
❯ wget -qO- "https://clarity.design/.netlify/functions/download-icon?set=core&shape=event-line"

<svg version="1.1" width="36" height="36"  viewBox="0 0 36 36" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <title>event-line</title>
    <path class="clr-i-outline clr-i-outline-path-1" d="M16.17,25.86,10.81,20.5a1,1,0,0,1,1.41-1.41L16.17,23l8.64-8.64a1,1,0,0,1,1.41,1.41Z"></path><path class="clr-i-outline clr-i-outline-path-2" d="M32.25,6H29V8h3V30H4V8H7V6H3.75A1.78,1.78,0,0,0,2,7.81V30.19A1.78,1.78,0,0,0,3.75,32h28.5A1.78,1.78,0,0,0,34,30.19V7.81A1.78,1.78,0,0,0,32.25,6Z"></path><path class="clr-i-outline clr-i-outline-path-3" d="M10,10a1,1,0,0,0,1-1V3A1,1,0,0,0,9,3V9A1,1,0,0,0,10,10Z"></path><path class="clr-i-outline clr-i-outline-path-4" d="M26,10a1,1,0,0,0,1-1V3a1,1,0,0,0-2,0V9A1,1,0,0,0,26,10Z"></path><rect class="clr-i-outline clr-i-outline-path-5" x="13" y="6" width="10" height="2"></rect>
    <rect x="0" y="0" width="36" height="36" fill-opacity="0"/>
</svg>%
```

And if you try to request `outline` only

```bash
❯ wget -qO- "https://clarity.design/.netlify/functions/download-icon?set=core&shape=event-outline"

404: Not Found%
```